File: java/src/main/java/com/fengtian/qianer/atcoder/abc/abc131/ABC131_B.java
package com.fengtian.qianer.atcoder.abc.abc131;

import java.util.Scanner;
import java.util.stream.IntStream;

public class ABC131_B {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        final int N = sc.nextInt();
        final int L = sc.nextInt();

        final int[] tastes = new int[N];
        for (int i = 0; i < N; i++) {
            tastes[i] = L + (i+1) - 1;
        }

        int x = -1;
        int min = Integer.MAX_VALUE;
        for (int i = 0; i < N; i++) {
            if (min > Math.abs(tastes[i])) {
                x = i;
                min = Math.abs(tastes[i]);
            }
        }

        tastes[x] = 0;

        System.out.println(IntStream.of(tastes).sum());
    }
}

File: kotlin/src/abc/abc131/ABC131_B.kt
package abc.abc131

fun main(args: Array<String>) {
    val (n, l) = readLine()!!.split(" ").map(String::toInt)
    val list = (1..n).map { l + it - 1 }.toList()
    val absMin = list.map(Math::abs).min()!!
    val relMin = if (list.min()!! > 0) absMin else -absMin
    println(list.sum() - relMin)
}